index.js

```diff
@@ -1,17 +1,22 @@
 'use strict';
 
-var HelloTransform = require('./lib/hello');
+var ThisFallbackPlugin = require('./lib/this-fallback-plugin');
 
 module.exports = {
   name: require('./package').name,
 
   included: function () {
-    // we have to wrap these in an object so the ember-cli
-    // registry doesn't try to call `new` on them (new is actually
-    // called within htmlbars when compiling a given template).
+    const plugin = ThisFallbackPlugin;
+    plugin.baseDir = () => {
+      return __dirname;
+    };
+    plugin.cacheKey = () => {
+      return 'ember-this-fallback';
+    };
```

Comment:
This stuff is not super well documented, but this `cacheKey` shenanigans makes the warning we talked about go away. 🤷‍♀️

Reply:
This makes sense. I think we probably will have at least one option once we implement Embroider support (`supportCustomResolver` or something like that). We can wait until we have options to care about caching for them obviously :)